src/librustc_span/source_map.rs

```diff
@@ -194,12 +209,25 @@ impl SourceMap {
         self.files.borrow().stable_id_to_source_file.get(&stable_id).map(|sf| sf.clone())
     }
 
-    fn next_start_pos(&self) -> usize {
-        match self.files.borrow().source_files.last() {
-            None => 0,
-            // Add one so there is some space between files. This lets us distinguish
-            // positions in the `SourceMap`, even in the presence of zero-length files.
-            Some(last) => last.end_pos.to_usize() + 1,
+    fn allocate_address_space(&self, size: usize) -> Result<usize, OffsetOverflowError> {
+        let size = u32::try_from(size).map_err(|_| OffsetOverflowError)?;
```

Comment:
I think you can do 

```suggestion
        // Add one so there is some space between files. This lets us distinguish
        // positions in the `SourceMap`, even in the presence of zero-length files.
        let size = u32::try_from(size + 1).map_err(|_| OffsetOverflowError)?;
```

here which avoids needing to `and_then()` the `checked_add` below.

Reply:
That addition could also overflow though.

Reply:
Yeah, that's true. That would imply `src` is `usize::max` long though. Wouldn't we already have OOM'd trying to read that in?

Reply:
Probably, but that fact isn't very local and would require a comment =P